src/libs/ReportPrimaryActionUtils.ts

```diff
@@ -0,0 +1,188 @@
+import type {OnyxCollection} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import CONST from '@src/CONST';
+import type {Policy, Report, Transaction, TransactionViolation} from '@src/types/onyx';
+import {isApprover as isApprovedMember} from './actions/Policy/Member';
+import {getCurrentUserAccountID} from './actions/Report';
+import {arePaymentsEnabled, getCorrectedAutoReportingFrequency, hasAccountingConnections, isAutoSyncEnabled, isPrefferedExporter} from './PolicyUtils';
+import {
+    isClosedReport,
+    isCurrentUserSubmitter,
+    isExpenseReport,
+    isHoldCreator,
+    isInvoiceReport,
+    isIOUReport,
+    isOpenReport,
+    isPayer,
+    isProcessingReport,
+    isReportApproved,
+    isSettled,
+} from './ReportUtils';
+import {getSession} from './SessionUtils';
+import {allHavePendingRTERViolation, isDuplicate, isOnHold as isOnHoldTransactionUtils, shouldShowBrokenConnectionViolationForMultipleTransactions} from './TransactionUtils';
+
+function isSubmitAction(report: Report, policy: Policy) {
+    const isExpense = isExpenseReport(report);
+    const isSubmitter = isCurrentUserSubmitter(report.reportID);
+    const isOpen = isOpenReport(report);
+    const isManualSubmitEnabled = getCorrectedAutoReportingFrequency(policy) === CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL;
+
+    return isExpense && isSubmitter && isOpen && isManualSubmitEnabled;
+}
+
+function isApproveAction(report: Report, policy: Policy, reportTransactions: Transaction[]) {
+    const isExpense = isExpenseReport(report);
+    const isApprover = isApprovedMember(policy, getCurrentUserAccountID());
+    const isApprovalEnabled = policy.approvalMode && policy.approvalMode !== CONST.POLICY.APPROVAL_MODE.OPTIONAL;
+
+    if (!isExpense || !isApprover || !isApprovalEnabled) {
+        return false;
+    }
+
+    const isOneExpenseReport = isExpense && reportTransactions.length === 1;
+    const isOnHold = reportTransactions.some(isOnHoldTransactionUtils);
+    const isProcessing = isProcessingReport(report);
+    const isOneExpenseReportOnHold = isOneExpenseReport && isOnHold;
+
+    if (isProcessing || isOneExpenseReportOnHold) {
```

Comment:
This(`isOneExpenseReportOnHold`) seems to be an incorrect condition to derive an approve action which caused an issue https://github.com/Expensify/App/issues/65873